### DISCUSSION TITLE
Escape arguments after `--`

### DISCUSSION THREAD

**Discussion by theofidry** [theofidry/console#243] [Ideas]
From @dantleech

one feature I'd love to see is using `--` to stop parsing args/options vs. disabling "validation" completely
e.g. `console --option=1 arg1 arg2 -- whatever --foo --bar hello`
and you'd be able to get the raw arguments after the `--` from the `Input`

**Comment by dantleech** [comment]
To provide an example of both this + sub-commands (imagining what [Castor](https://github.com/jolicode/castor) could facilitate):

```
castor --working-dir=foobar qa --changed-files phpstan -- --debug path/to/files
```

This syntax is supported by Clap, Kong etc. and also my ill-fated attempt at writing an argument parser: https://github.com/php-tui/cli-parser

**Reply by theofidry** [reply]
I totally see the use case, but I'm wondering code-wise how you would consume this extra token, would you just get an extra string, or would you want the parsed raw tokens that you can append?

**Reply by dantleech** [reply]
- Example with Clap: https://docs.rs/clap/latest/clap/struct.Command.html#method.allow_missing_positional
- As "passthrough" in Kong: https://github.com/alecthomas/kong?tab=readme-ov-file#supported-tags

In both examples it's `string[]` - which makes sense as it's essentially the rest of "`$argv`" and there would be a diff between `-- "foobar barfoo"` and `-- foobar barfoo`
